### DISCUSSION TITLE
Custom modifications with diagnostic ions at MS2

### DISCUSSION THREAD

**Discussion by Hannahluise** [vdemichev/DiaNN#1051] [Q&A]
Thank you a lot for the great work with DIA-NN!

I am just starting with the software and have a specific question regarding PTMs with diagnostic ions. I would like to add a custom PTM via the command-line option. Upon CID, a diagnostic ion will be produced from modified peptides that can be detected at MS2 level. 

Is there any way,  I could incorporate this into my custom modification settings, using the mod option, to enhance my identification reliability for said modification?

I would really appreciate everyone's support and wish you a great weekend. 

Hannah Voß

**Comment by vdemichev** [comment]
Hi Hannah,

Possible in a 'hacky' way. So what you could do:
- Declare the modification as a --fixed-mod or --var-mod, generate an in silico predicted spectral library, DIA-NN will save a .predicted.speclib file.
- Convert this .predicted.speclib file to .tsv (can be large, ~10x larger than .predicted.speclib). If this becomes too large to handle on your PC, can carry out the whole procedure separately for different precursor mass ranges (e.g. 400-500, 500-600, etc). Also, makes sense to first try with no or minimal number of variable modifications, precursor charges 2-3 and precursor mass range restricted to the actual range of the DIA acquisition scheme.
- Edit the .tsv to (i) adjust the mass of each fragment including the modification to account for the loss of the diagnostic ion and to (ii) add an extra fragment ion with the mass of the diagnostic one, annotating it as b1 with charge 1 and no loss (does not matter that this is an incorrect annotation).
- Analyse with this edited library (or with multiple --lib commands, if you did this procedure separately for several precursor mass ranges). Enable peptidoform scoring if this is important.

Best,
Vadim
